src/cephadm/cephadm

```diff
@@ -3182,6 +3187,36 @@ class CephContainer:
         self.memory_request = memory_request
         self.memory_limit = memory_limit
 
+    @property
+    def cname(self) -> str:
+        """
+        podman adds the current container name to the /etc/hosts
+        file. Turns out, python's `socket.getfqdn()` differs from
+        `hostname -f`, when we have the container names containing
+        dots in it.:
+
+        # podman run --name foo.bar.baz.com ceph/ceph /bin/bash
+        [root@sebastians-laptop /]# cat /etc/hosts
+        127.0.0.1   localhost
+        ::1         localhost
+        127.0.1.1   sebastians-laptop foo.bar.baz.com
+        [root@sebastians-laptop /]# hostname -f
+        sebastians-laptop
+        [root@sebastians-laptop /]# python3 -c 'import socket; print(socket.getfqdn())'
+        foo.bar.baz.com
+
+        Fascinatingly, this doesn't happen when using dashes.
+        """
+        return self._cname.replace('.', '-')
```

Comment:
Just curious, did you look at `socket.getfqdn()` source to see what it is doing exactly?

Reply:
Nevermind, it is actually documented:
```
To find the fully qualified name, the hostname returned by gethostbyaddr() is checked,
followed by aliases for the host, if available. The first name which includes a period is selected.
In case no fully qualified domain name is available, the hostname as returned by gethostname()
is returned.
```

Reply:
I still don't get why `hostname -f` returns something different here.